packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -109,7 +109,8 @@ export abstract class AbstractBackgroundJobProcessor<
     this.runningPromises = []
   }
 
-  public getJobCount(): Promise<number> {
+  public async getJobCount(): Promise<number> {
+    await this.startIfNotStarted()
```

Comment:
can we do an explicit synchronous `this.isStarted` check first, to avoid introducing one more promise and one more tick on every call?

Reply:
`startIfNotStarted` is already doing it, do you think doing it also here makes sense?

Reply:
I have just seen your updated version, sure, yeah, that is better

Reply:
note that there is still overhead in using Promise.resolve over purely synchronous code, and it also relies on microtasks to work, but it is, indeed, faster than async functions, so it is fine to keep it like this

Reply:
I completely agree with you and thank you so much for giving the extended info, I really like and appreciate your comments/explanation, it is really useful to learn ❤️